apple/LayoutReanimation/REASwizzledUIManager.mm

```diff
@@ -324,4 +348,32 @@ - (RCTViewManagerUIBlock)reanimated_uiBlockWithLayoutUpdateForRootView:(RCTRootS
   };
 }
 
+- (void)reanimated_addUIBlock:(RCTViewManagerUIBlock)block
+{
+  [self.batchObserver onNewBatchBlockQueued];
+  [self reanimated_addUIBlock:block];
+}
+
+- (void)reanimated_prependUIBlock:(RCTViewManagerUIBlock)block
+{
+  [self.batchObserver onNewBatchBlockQueued];
+  [self reanimated_prependUIBlock:block];
+}
+
+- (void)reanimated_flushUIBlocksWithCompletion:(void (^)(void))completion
+{
+  NSNumber *batchId = [self.batchObserver batchWillFlush];
+  NSMutableArray<RCTViewManagerUIBlock> *pendingUIBlocks = [self valueForKey:@"_pendingUIBlocks"];
```

Comment:
I think there is no need to check _pendingUIBlocks here. We already rely on a fact that blocks can only be added in addUIBlock and prependUIBlock. We should just keep track on whether one of these two methods has been called before the last flush